### PR TITLE
Angular Stories

### LINKED ISSUE #1852: Add Modals Component - Angular NDS

### LINKED ISSUE #1854: Add Masthead Component - Angular NDS

### LINKED ISSUE #1856: Add Pagination Component - Angular NDS

### PR DESCRIPTION
**Pull request by guern1kn**
## What does this PR do?
Adds component stories to Angular - Modals, Masthead, Pagination
Adds actions for Modals - Angular & React

### Associated Issue 
Fixes #1852 
Fixes #1854 
Fixes #1856 

### Documentation
Modals
 - [x] Update Spark Docs Angular

Masthead
 - [x] Update Spark Docs Angular

Pagination
 - [x] Update Spark Docs Angular

### Code
Modals
 - [x] Build Component in Spark Angular
 - [x] Unit Testing in Spark Angular with `gulp test-angular` (100% coverage, 100% passing)

Masthead
 - [x] Build Component in Spark Angular
 - [x] Unit Testing in Spark Angular with `gulp test-angular` (100% coverage, 100% passing)

Pagination
 - [x] Build Component in Spark Angular
 - [x] Unit Testing in Spark Angular with `gulp test-angular` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
Modals
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

Masthead
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

Pagination
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### Screenshots
Modals
image
image
image

Masthead - Default
image
image
image

Masthead - Extended
image
image
image

Pagination
image
image
image

### CODE REVIEW

File: angular/projects/spark-angular/src/lib/components/sprk-modal/sprk-modal.stories.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      (confirmClick)="confirmClick($event)"
+    >
+      This is some modal content.
+    </sprk-modal>
+  `,
+  props: {
+    choiceVisible: true,
+    toggleChoiceModal: (event: any) => {
+      event.preventDefault();
+      alert('cancel click');
```
Comment (afebbraro) [inline angular/projects/spark-angular/src/lib/components/sprk-modal/sprk-modal.stories.ts]: It feels unexpected to click the modal in storybook and get a browser alert saying "cancel click". Rob and I have chatted about this experience after we saw the react modal implemented that way but with things being so busy haven't been able to chat about it with everyone to get opinions. Should we bring this up now?

File: angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.stories.ts [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+      },
+      {
+        text: 'My Account',
+        href: '#nogo',
+        active: false,
+        leadingIcon: 'user',
+        subNav: [
+          {
+            text: 'Change Username',
+            href: '/links'
```
Comment (afebbraro) [inline angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.stories.ts]: What happens when this link with the value of '/links' is clicked in storybook? I don't think we have a route there.

Comment (guern1kn) [reply angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.stories.ts]: It doesn't do anything or navigate anywhere. But I can clean these up to be #nogo if that's preferable.

File: angular/projects/spark-angular/src/lib/components/sprk-modal/sprk-modal.stories.ts (hunk 1 above)
Comment (guern1kn) [reply angular/projects/spark-angular/src/lib/components/sprk-modal/sprk-modal.stories.ts]: Yeah, I thought that maybe was odd too. Happy to chat about it before we merge.

**Pull request merged**